OasisDEX/oasis-borrow
# Ades/claim curve

# [Claim LDO rewards](https://app.shortcut.com/oazo-apps/story/3895/add-a-banner-for-claiming-lido-rewards)
  
Adds functionality to claim LDO rewards as part of CRVV1ETHSTETH-A ilk.

<img width="1121" alt="image" src="https://user-images.githubusercontent.com/3783198/165792491-ef706f0f-8e23-4659-abcf-d238fe9757d0.png">

## Changes 👷‍♀️

For Ilks that support rewards (only CRVV1ETHSTETH-A right now), the box is displayed.  The user can click the button to claim their rewards.

Tech notes:

- New integrations with smart contracts:
  - Lido Curve liquidity farming reward, taken from https://docs.lido.fi/deployed-contracts
  - LDO ERC20, also from https://docs.lido.fi/deployed-contracts
  - Maker Cropjoin from makerdao chainlog
- inside `/features/bonus`:
  - New pipe bonusPipe, with a makerprotocol-specific adapter that handles the blockchain stuff.  Pipe is independent of  manage borrow pipe.
  - BonusContainer that consumes the pipe and provides the UI.
- Updates erc20 abi to include decimals (taken from second google result https://gist.github.com/veox/8800debbf56e24718f9f483e1e40c35c)
  
## How to test 🧪

- find a CRVV1ETHSTETH-A vault from https://tracker-vaults.makerdao.network/collateral/CRVV1ETHSTETH-A
- click through to ethx and cmd-f for `open`, then find the CDP ID.
- use hardhat and steal the proxy for the CDP ID using hardhat task `npx hardhat get-proxy --proxy <proxy_addr> --network local` where <proxy_addr> is the proxy of the CDP ID.  Your first Hardhat wallet will now have the proxy.
- oasis-borrow at hardhat and use the first hardhat address.
- Navigate to the CDP ID (it should be a CRVV1ETHSTETH-A position)
- You should be able to see the amount of unclaimed LDO, claim the LDO, and see the same amount get transferred to the wallet.
